app/src/main/java/com/example/universitieslist/data/remote/repository/UniversityRepository.kt

```diff
@@ -4,5 +4,5 @@ import com.example.universitieslist.data.model.UniversityResponse
 import kotlinx.coroutines.flow.Flow
 
 interface UniversityRepository {
-    suspend fun getUniversities(country: String): Flow<List<UniversityResponse>>
+    suspend fun getUniversities(country: String): Flow<Result<List<UniversityResponse>?>>
```

Comment:
Oh! I did not say you should use `Flow` and `Result` together. Since you're using Flow in the repository, when calling the function in the VM, that's where I said you will call the `onError{}` block.